content/building-blocks/automate-and-execute-your-work/reproducible-work/google_cloud_docker.md

```diff
@@ -0,0 +1,312 @@
+
+{{% codeblock %}}
+```bash
+$ sudo gcsfuse -o allow_other your-bucket volume_path/new_dir/
+```
+{{% /codeblock %}}
+
+This code will tell GCSFuse to syncronize your new directory within the path with your bucket and allow your container to access it. After this you just have to store any output produced in your environment in this new directory that you just created in your instance a few moments ago (Referred in the previous codeblock as "new_dir") and it will be inmediatly available to you in your GCS bucket.
+
+{{% tip %}}
+
+Bear in mind that the absolute path to your bucket-sincronized directory is not the same for your instance and for your container given that containers have independent file systems. However its relative path will be the same with respect to that where the Docker volume is mounted.
+
+{{% /tip %}}
+
+
+### Step 4: Expose the port where Jupyter is running to access the environment from your computer
+
+At this point we need to make our instance accesible from the outside. The first element we need for this task is know in which port of our instance is Jupyter running. To know this we just have to take a closer look to the last line of the output shown in the command line after we executed `docker compose up`. At the beggining of the line you should see an address like the one shown below, from there you interested in the four numbers underlined in bluecoming after the internal IP and before the slash. These correspond with the port where Jupyter is running in your instance.
+
+<p align = "center">
+<img src = "../img/find_port.png" width="750">
+<figcaption> Underlined in blue: The port that you need to expose. </figcaption>
+</p>
+
+{{% warning %}}
+
+Note that the port in which your Jupyter environment is running (i.e the four numbers that correspond to those underlined in blue in the previous image) will probably be different to the ones shown in the image above.
+
+{{% /warning %}}
+
+{{% tip %}}
+
+You can actually choose in which port do you want Jupyter to be executed. In the context of this building block this should be done by modifying the docker-compose, where the instructions on how Docker should set up Jupyter for your environment are contained. [Learn more about docker compose.](https://docs.docker.com/compose/)
+
+{{% /tip %}}
+
+Now you should go to back to the Google cloud console and click the button "Set up firewall rules" which is below your instances list. Then click again on the "Create firewall rule" option located towards the upper side of the screen. There you must:
+
+1. Add a name of your choice for the rule (e.g. Jupyter_access_rule)
+2. Make sure that traffic direction is set on "Ingress"
+3. Introduce "0.0.0.0/0" in the field "Source IPv4 ranges"
+4. In the section "Protocols and ports" check "TCP" and then in the box below introduce the four number which correspond to the port where Jupyter is being executed in your instance.

```

Comment:
Possible typo: `port number` instead of `four number`

Reply:
I wanted to make it easier to identify the port number for users for which this concept may be new so I decided to stress that it is 4-digit long. In any case there is a typo present as there should be an extra plural on "number" so the idea is correctly expressed. All in all I've decided to reformulate that fourth step to: 

_"4. In the section "Protocols and ports" check "TCP" and then in the box below introduce the four-digit number that identifies the port where Jupyter is being executed in your instance."_

This will be reflected in the next update of the pull request.